core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -722,16 +720,25 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         if (query == null) {
             return null;
         }
+
+        //save the BoostQuery wrapped structure if present
+        List<Float> boosts = new ArrayList<>();
+        while(query instanceof BoostQuery) {
+            BoostQuery boostQuery = (BoostQuery) query;
+            boosts.add(boostQuery.getBoost());
+            query = boostQuery.getQuery();
+        }
```

Comment:
I dunno, I thought it's better not to replicate the internal lucene behaviour? (the fact that it multiplies boosts).